src/_pytest/_io/terminalwriter.py

```diff
@@ -180,7 +180,7 @@ def _write_source(self, lines: Sequence[str], indents: Sequence[str] = ()) -> No
 
     def _highlight(self, source: str) -> str:
         """Highlight the given source code if we have markup support."""
-        if not self.hasmarkup:
+        if not self.hasmarkup and not self.code_highlight:
```

Comment:
`self.code_highlight` should be initialized in `__init__` I think (both for correctness and clarity).

Reply:
Fixed.